beacon-chain/slasher/detect_attestations.go

```diff
@@ -71,6 +81,69 @@ func (s *Service) detectSlashableAttestations(
 	return s.serviceCfg.Database.SaveLatestEpochAttestedForValidators(ctx, validatorIndices, args.currentEpoch)
 }
 
+// Check for attester slashing double votes by looking at every single validator index
+// in each attestation's attesting indices and checking if there already exist records for such
+// attestation's target epoch. If so, we append a double vote slashing object to a list of slashings
+// we return to the caller.
+func (s *Service) checkDoubleVotes(
```

Comment:
Im afraid `detect_attestations.go ` will get too big, I'd consider breaking them down into smaller files:
`detech_double_votes.go`, `detech_surround_votes.go`... etc

Reply:
I can do this in a separate PR, im afraid the diff will then be too big in this PR